src/functions/welcome/welcome.go

```diff
@@ -0,0 +1,66 @@
+package welcome
+
+import (
+	"encoding/json"
+	"fmt"
+	"html"
+	"net/http"
+	"strings"
+	"os"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+// Data struct from Pubsub
+type PubSubMessage struct {
+	Data []byte `json:"data"`
+}
+
+type Payload struct {
+	UserId  string `json:"user_id"`
+	Username string `json:"username"`
+	ServerId string `json:"server_id"`
+	ServerName string `json:"server_name"`
+	AvatarUrl string `json:"avatar_url"`
+	ChannelId string `json:"channel_id"`
+}
+
+func GreetingNewMember(w http.ResponseWriter, r *http.Request) {
+	//Parse body to get Payload
+	var payload = Payload{}
+	err := json.NewDecoder(r.Body).Decode(&payload)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if len(payload.UserId) == 0 
+	or len(payload.Username) == 0 
+	or len(payload.ServerName) == 0 
+	or len(payload.AvatarUrl) == 0 {
+		panic("Missing data in payload")
+	}
+
+	// Instantiate Discord bot
+	dg, err := discordgo.New("Bot " + os.Getenv("DISCORD_TOKEN"))
+	if err != nil {
+		fmt.Println("error creating Discord session: ", err)
+	}
+
+	// Get the channel
+	channel, err := dg.Channel(payload.ChannelId)
+	if err != nil {
+		panic("Unauthorized to create the connection. Verify Discord Token or ChannelID")
+	}
+
+	//Build a string efficiently with strings.Builder
+	var message strings.Builder
+	message.WriteString("Hey {}, welcome to {}!\n", payload.Username, payload.ServerName)
+	message.WriteString("Check out the <#{}> and have fun!.\n", payload.ServerId)
```

Comment:
It's not `ServerId` that we want

Reply:
Sorry, that one was actually a mistake of my part! I will fix it